cmd/image-builder/upload.go

```diff
@@ -104,10 +100,11 @@ func uploaderForCmdAWS(cmd *cobra.Command) (cloud.Uploader, error) {
 		}
 	}
 	if len(missing) > 0 {
-		return nil, &MissingUploadConfigError{
-			missing:    missing,
-			allMissing: len(missing) == len(requiredArgs),
+		if len(missing) == len(requiredArgs) {
+			return nil, ErrMissingUploadConfigAll
```

Comment:
This change makes the tests fail and is a UX regression, the tests check that the user gets a message what is missing so that they have the information right away. By just returning a "naked" ErrMissingUploadConfigAll that information is no longer transmitted to the user.